regression/contracts/enforce-unknown-function/test.desc

```diff
@@ -0,0 +1,12 @@
+CORE
+main.c
+--enforce-contract goo
+^Invariant check failed$
+^Condition: Precondition$
+^Reason: all_functions_found\(to_enforce\)$
```

Comment:
Now throwing an exception instead of using a precondition. The guidelines for error handling are: handle incorrect user input gracefully, throwing informative exceptions from `util/exception_utils.h`, use INVARIANT and PRECONDITION for internal checks that shall never fail.

Reply:
Sounds good